### PR TITLE
Whitespace after the caret should not be removed When inserting slashes on ENTER in XML doc comments

### LINKED ISSUE #2091: Incorrect behavior of ENTER in XML documentation comment when whitespace is to the right of the editor caret
**Repro Steps:**

* Enter the following code, and position the caret at the location indicated by `|`. *Note that the code block is indented.*
```C#
            /// <summary>
            /// foo| bar
            /// </summary>
            public void Foo() { }
```

* Press Enter

**Expected Result:**
```C#
        /// <summary>
        /// foo
        /// |bar
        /// </summary>
        public void Foo() { }
```

**Actual Result:**
```C#
        /// <summary>
        /// foo
       /// | bar
        /// </summary>
        public void Foo() { }
```

### PR DESCRIPTION
**Pull request by DustinCampbell**
Fixes #2091 

Reviewers: @CyrusNajmabadi, @Pilchie, @jasonmalinowski, @rchande, @dpoeschl, @balajikris, @jmarolf, @davkean, @brettfo

### CODE REVIEW

**Comment by sharwell** [comment]
This addresses only one of two spacing problems described in #2091. I'm not sure I agree with the commit message; the issue specifically indicates that whitespace after the caret *should* be removed in this case.

File: src/EditorFeatures/CSharpTest/DocumentationComments/DocumentationCommentTests.cs [hunk 1]
```diff
@@ -1084,6 +1084,36 @@ static void Main(string[] args)
...
+    {
+    }
+}";
+
+            const string expected =
+@"class C
+{
+    /// <summary>
+    /// hello
+    /// $$ world
```
Comment (sharwell) [inline src/EditorFeatures/CSharpTest/DocumentationComments/DocumentationCommentTests.cs]: :exclamation: I expected the following here:

```csharp
    /// $$world
```

File: src/EditorFeatures/VisualBasicTest/DocumentationComments/DocumentationCommentTests.vb [hunk 2]
```diff
@@ -586,6 +586,31 @@ $$''' <summary>
...
+    ''' </summary>
+    Sub M()
+    End Sub
+End Class
+"
+            Const expected = "
+Class C
+    ''' <summary>
+    ''' hello
+    ''' $$ world
```
Comment (sharwell) [inline src/EditorFeatures/VisualBasicTest/DocumentationComments/DocumentationCommentTests.vb]: :exclamation: I expected the following here:

```vb
    ''' $$world
```

Comment (DustinCampbell) [inline src/EditorFeatures/VisualBasicTest/DocumentationComments/DocumentationCommentTests.vb]: You're right. I didn't look carefully at the bug. Thanks for checking!

File: src/EditorFeatures/CSharpTest/DocumentationComments/DocumentationCommentTests.cs (hunk 1 above)
Comment (DustinCampbell) [inline src/EditorFeatures/CSharpTest/DocumentationComments/DocumentationCommentTests.cs]: You're right. I didn't look carefully at the bug. Thanks for checking!

**Comment by DustinCampbell** [comment]
OK. Should be fixed appropriately now and code to insert exterior trivia is unified with the fix for #2090.

**Comment by DustinCampbell** [comment]
commit message updated. Thanks @sharwell!

**Comment by sharwell** [comment]
:bulb: This change would benefit from the following additional test. The following case currently passes as described in Visual Studio 2013 and earlier.

**Before:**

```csharp
/// <summary>
///     hello$$ world
/// </summary>
```

**After:**

```csharp
/// <summary>
///     hello
///     $$world
/// </summary>
```

**Comment by DustinCampbell** [comment]
Thanks Sam, I've already got a that test in my fix for #2108 (incoming!)

**Comment by sharwell** [comment]
:+1:

**Comment by DustinCampbell** [comment]
@sharwell -- TYVM for your help here.

**Pull request merged**